File: php/stack.php
<?php

class Stack {
    public $top;
    public $stack = array();

    function __construct() {
        $this->top = -1;
    }

    function isEmpty() {
        if ($this->top == -1) {
            echo "The stack is empty. \n";
            return true;
        }
        echo "The stack is not empty. \n";
        return false;
    }

    function size() {
        $size = $this->top + 1;
        echo "The size of the stack is " . $size . " elements. \n";
    }

    function topElement() {
        if ($this->isEmpty()) {
            echo "There is no top element in the stack. \n";
        } else {
            $topElement = $this->stack[$this->top];
            echo "The top element is: " . $topElement . ". \n";
        }
    }

    function push($element) {
        $this->stack[++$this->top] = $element;
        echo $element . " was added to the stack. \n";
    }

    function pop() {
        if ($this->isEmpty()) {
            echo "Nothing to pop - the stack is empty. \n";
        } else {
            $element = $this->stack[$this->top--];
            echo $element . " was removed from the stack. \n";
        }
    }
}

$myStack = new Stack();

$myStack->size();
$myStack->topElement();

$myStack->push(10);
$myStack->push(20);
$myStack->push(30);

$myStack->size();
$myStack->topElement();

$myStack->pop();
$myStack->size();
$myStack->topElement();
?>

File: python/stack.py
from contextlib import nullcontext


class Stack:
    def __init__(self):
        self.top = -1
        self.stack = []

    def is_empty(self):
        if self.top == -1:
            print('The stack is empty.')
            return True
        else:
            print('The stack is not empty.')
            return False

    def size(self):
        size = self.top + 1
        print(f'The size of the stack is {size} elements.')

    def top_element(self):
        if self.is_empty():
            print('There is no top element in the stack.')
        else:
            top_element = self.stack[self.top]
            print(f'The top element is: {top_element}.')
    
    def push(self, element):
        self.top = self.top + 1
        self.stack.insert(self.top, element)
        print(f'{element} was added to the stack.')

    def pop(self):
        if self.is_empty():
            print('Nothing to pop - the stack is empty.')
        else:
            element = self.stack[self.top]
            self.top = self.top - 1
            print(f'{element} was removed from the stack.')

myStack = Stack()

myStack.size()
myStack.top_element()

myStack.push(10)
myStack.push(20)
myStack.push(30)

myStack.size()
myStack.top_element()

myStack.pop()
myStack.size()
myStack.top_element()
    
